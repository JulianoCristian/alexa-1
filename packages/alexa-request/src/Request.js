import { Request as Constants } from 'alexa-constants';

export default class Request {
  static session = (...args) => new Request().session(...args);
  static intent = (...args) => new Request().intent(...args);
  static launchRequest = (...args) => new Request().launchRequest(...args);
  static sessionEndedRequest = (...args) => new Request().sessionEndedRequest(...args);

  constructor(state = {}) {
    this.state = state;
  }

  session(session) {
    return new Request({
      ...this.state,
      session: {
        ...this.state.session,
        ...session
      }
    });
  }

  intent(name, slots = {}) {
    const slotData = Object.entries(slots).reduce((state, [name, value]) => ({ ...state, [name]: { name, value } }), {});
    return new Request({
      ...this.state,
      request: {
        type: Constants.Intent,
        intent: {
          name: name,
          ...(Object.keys(slotData).length ? { slots: slotData } : null)
        }
      }
    });
  }

  launchRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.Launch
      }
    });
  }

  sessionEndedRequest() {
    return new Request({
      ...this.state,
      request: {
        type: Constants.SessionEnded
      }
    });
  }

  build() {
    return { ...this.state };
  }
}
