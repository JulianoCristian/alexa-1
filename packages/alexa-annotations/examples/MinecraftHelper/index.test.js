import Request from 'alexa-request';
import MinecraftHelper from './index';

test('LaunchRequest', () => {
  const event = Request.launchRequest().build();
  return expect(MinecraftHelper(event)).resolves.toMatchSnapshot();
});

test('RecipeIntent', () => {
  const event = Request.intent('RecipeIntent', { Item: 'Farmland' }).build();
  return expect(MinecraftHelper(event)).resolves.toMatchSnapshot();
});

test('Unknown recipe', () => {
  const event = Request.intent('RecipeIntent', { Item: 'Coffee' }).build();
  return expect(MinecraftHelper(event)).resolves.toMatchSnapshot();
});

test('No recipe', () => {
  const event = Request.intent('RecipeIntent').build();
  return expect(MinecraftHelper(event)).resolves.toMatchSnapshot();
});
