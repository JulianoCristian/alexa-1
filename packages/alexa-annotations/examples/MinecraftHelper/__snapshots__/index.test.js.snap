// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchRequest 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "Welcome to the Minecraft Helper. You can ask a question like, what's the recipe for a chest? ... Now, what can I help you with.",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "For instructions on what you can say, please say help me.",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`No recipe 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "I'm sorry, I currently do not know that recipe. What else can I help with?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "What else can I help with?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;

exports[`RecipeIntent 1`] = `
Object {
  "response": Object {
    "card": Object {
      "content": "Farmland can be created by plowing the land with a hoe.",
      "title": "Recipe for Farmland",
      "type": "Simple",
    },
    "outputSpeech": Object {
      "text": "Farmland can be created by plowing the land with a hoe.",
      "type": "PlainText",
    },
    "shouldEndSession": true,
  },
  "version": "1.0",
}
`;

exports[`Unknown recipe 1`] = `
Object {
  "response": Object {
    "outputSpeech": Object {
      "text": "I'm sorry, I currently do not know the recipe for Coffee. What else can I help with?",
      "type": "PlainText",
    },
    "reprompt": Object {
      "outputSpeech": Object {
        "text": "What else can I help with?",
        "type": "PlainText",
      },
    },
    "shouldEndSession": false,
  },
  "version": "1.0",
}
`;
